sbi/inference/abc/smcabc.py

```diff
@@ -0,0 +1,487 @@
+from __future__ import annotations
+
+from pyro.distributions.empirical import Empirical
+from sbi.inference.abc.abc_base import ABCBASE
+from typing import Callable, Optional, Tuple, Union
+
+import torch
+from torch import Tensor, ones, tensor
+from torch.distributions import Distribution, Multinomial, MultivariateNormal
+from pyro.distributions import Uniform
+from numpy import ndarray
+import logging
+
+
+class SMCABC(ABCBASE):
+    def __init__(
+        self,
+        simulator: Callable,
+        prior: Distribution,
+        x_o: Union[Tensor, ndarray],
+        distance: Union[str, Callable] = "l2",
+        num_workers: int = 1,
+        simulation_batch_size: int = 1,
+        show_progress_bars: bool = True,
+        kernel: Optional[str] = "gaussian",
+        algorithm_variant: str = "C",
```

Comment:
Does it make sense to break them into separate classes? (Ok to do in a future PR if too much work).

Reply:
I thought about this, but the differences are very small so I dont think that this would make sense.